multipass-addressing/src/main/scala/com/precisely/bigdata/sample/spark/MultipassAddressing.scala

```diff
@@ -0,0 +1,198 @@
+      .select("*", "addressing_geocode_result.*").drop("addressing_geocode_result")
+      // Write the dataframe to the specified output folder as csv file
+      .write.mode(SaveMode.Overwrite).option("header", "true").csv(outputDirectory)
+  }
+}
+
+/*
+ * A custom Executor that contains the logic to perform verify and then multipass geocode
+ */
+class CustomExecutor extends AddressingExecutor {
+
+  import com.precisely.addressing.v1.model.Response
+  import com.precisely.addressing.v1.{Addressing, Preferences}
+
+  // set preference to return all fields and Match Mode to Relaxed
+  private val relaxedGeocoderPreferences: Preferences = new PreferencesBuilder()
+    .withReturnAllInfo(true)
+    .withMatchMode("RELAXED")
+    .build()
+
+  override def execute(input: RequestInput, preferences: Option[Preferences], addressing: Addressing): Response = {
+
+    val verifyResponse = addressing.verify(input.requestAddress(), preferences.orNull)
+
+    val inputRequest: RequestAddress = input.requestAddress()
+
+    var multiLineRequest: RequestAddress = inputRequest
+
+    if (CollectionUtils.isNotEmpty(verifyResponse.getResults)) {
+      val verifyAddress = verifyResponse.getResults.get(0).getAddress
+
+      //  create multiLine request from verify Response
+      multiLineRequest = new RequestAddress()
+      multiLineRequest.setAddressLines(getValidAddressLine(verifyResponse.getResults.get(0).getAddressLines))
+      multiLineRequest.setStreet(verifyAddress.getStreet)
+      multiLineRequest.setAddressNumber(verifyAddress.getAddressNumber)
+      multiLineRequest.setCity(verifyAddress.getCity.getLongName)
+      multiLineRequest.setPostalCode(verifyAddress.getPostalCode)
+      multiLineRequest.setCountry(verifyAddress.getCountry.getIsoAlpha3Code)
+
+    }
+
+    val multiLineResponse = addressing.geocode(multiLineRequest, relaxedGeocoderPreferences)
+    if (CollectionUtils.isNotEmpty(multiLineResponse.getResults) && isAddressLevelMatch(multiLineResponse.getResults.get(0))) {
+      return multiLineResponse
+    }
+
+    val singleLine: String = if (CollectionUtils.isNotEmpty(verifyResponse.getResults)) {
+      for (addressLine <- 0 to getValidAddressLine(verifyResponse.getResults.get(0).getAddressLines).size() - 1) {
+        verifyResponse.getResults.get(0).getAddressLines.get(addressLine) + " "
+      }
+      verifyResponse.getResults.get(0).getAddress.getFormattedAddress + " " + verifyResponse.getResults.get(0).getAddress.getCountry.getIsoAlpha3Code
+    } else {
+      inputRequest.getStreet + " " +
+        inputRequest.getCity + " " +
+        inputRequest.getAdmin1 + " " +
+        inputRequest.getPostalCode
+    }
+
+    val singleLineRequest: RequestAddress = new RequestAddress()
+    singleLineRequest.setAddressLines(Collections.singletonList(singleLine))
+
+    val singleLineResponse = addressing.geocode(singleLineRequest, relaxedGeocoderPreferences)
```

Comment:
COMMENT: Perform the single line geocode and return the result if it passes our acceptance criteria